metadata-ingestion/tests/unit/test_mapping.py

```diff
@@ -231,3 +232,43 @@ def test_operation_processor_advanced_matching_tags():
     tag_aspect: GlobalTagsClass = aspect_map["add_tag"]
     assert len(tag_aspect.tags) == 1
     assert tag_aspect.tags[0].tag == "urn:li:tag:case_4567"
+
+
+def test_operation_processor_institutional_memory():
+    raw_props = {
+        "documentation_link": "test.com",
+    }
+    processor = OperationProcessor(
+        operation_defs={
+            "documentation_link": {
+                "match": ".*",
+                "operation": "add_doc_link",
+                "config": {"link": "{{ $match }}", "description": "test"},
+            },
+        },
+    )
+    aspect_map = processor.process(raw_props)
+    assert "add_doc_link" in aspect_map
+
+    doc_link_aspect: InstitutionalMemoryClass = aspect_map["add_doc_link"]
+
+    assert doc_link_aspect.elements[0].url == "test.com"
+    assert doc_link_aspect.elements[0].description == "test"
+
+
+def test_operation_processor_institutional_memory_no_description():
+    raw_props = {
+        "documentation_link": "test.com",
+    }
+    processor = OperationProcessor(
+        operation_defs={
+            "documentation_link": {
+                "match": ".*",
```

Comment:
can we also test that match param works correctly?

Reply:
something similar to the test case here https://github.com/datahub-project/datahub/blob/5b9fd977eb60d4a1a41d6aa7cadfb7bce01a93ab/metadata-ingestion/tests/integration/dbt/test_dbt.py#L62

e.g. match=`(.+)(?:#.*)?` and raw props =`http://example.com/my-page#stuff-to-ignore` should produce an aspect with link = `http://example.com/my-page`